pkg/cvo/availableupdates.go

```diff
@@ -176,29 +182,59 @@ func checkForUpdate(clusterID string, proxyURL *url.URL, upstream, channel strin
 	if len(upstream) == 0 {
 		return nil, fmt.Errorf("no upstream URL set for cluster version")
 	}
-	return cincinnati.NewClient(uuid, proxyURL).GetUpdates(upstream, channel, currentVersion)
+	return cincinnati.NewClient(uuid, proxyURL, tlsConfig).GetUpdates(upstream, channel, currentVersion)
 }
 
 // getHTTPSProxyURL returns a url.URL object for the configured
 // https proxy only. It can be nil if does not exist or there is an error.
-func (optr *Operator) getHTTPSProxyURL() (*url.URL, error) {
+func (optr *Operator) getHTTPSProxyURL() (*url.URL, string, error) {
 	proxy, err := optr.proxyLister.Get("cluster")
 
 	if errors.IsNotFound(err) {
-		return nil, nil
+		return nil, "", nil
 	}
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	if &proxy.Spec != nil {
 		if proxy.Spec.HTTPSProxy != "" {
 			proxyURL, err := url.Parse(proxy.Spec.HTTPSProxy)
 			if err != nil {
-				return nil, err
+				return nil, "", err
 			}
-			return proxyURL, nil
+			return proxyURL, proxy.Spec.TrustedCA.Name, nil
 		}
 	}
-	return nil, nil
+	return nil, "", nil
+}
+
+func (optr *Operator) getTLSConfig(cmNameRef string) (*tls.Config, error) {
+	cm, err := optr.cmManagedLister.Get(cmNameRef)
+
+	if errors.IsNotFound(err) {
```

Comment:
can make this function fail on not found.... ?
what the effects of doing that..?

Reply:
I see your point.  Since we are checking if `cmNameRef` is not empty if the ConfigMap isn't found that is probably not a good thing.  

The latest commit would `return err`:
https://github.com/openshift/cluster-version-operator/pull/226/files#diff-78f2af341fa49292dd6930f378018867R49